pkg/xds/generator/egress/external_services_generator.go

```diff
@@ -0,0 +1,195 @@
+package egress
+
+import (
+	"sort"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	xds_context "github.com/kumahq/kuma/pkg/xds/context"
+	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
+	envoy_clusters "github.com/kumahq/kuma/pkg/xds/envoy/clusters"
+	envoy_listeners "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+	"github.com/kumahq/kuma/pkg/xds/envoy/tls"
+)
+
+type ExternalServicesGenerator struct {
+}
+
+// Generate will generate envoy resources for one, provided in ResourceInfo mesh
+func (g *ExternalServicesGenerator) Generate(
+	_ xds_context.Context,
+	info *ResourceInfo,
+) (*core_xds.ResourceSet, error) {
+	resources := core_xds.NewResourceSet()
+
+	apiVersion := info.Proxy.APIVersion
+	endpointMap := info.EndpointMap
+	destinations := g.buildDestinations(info.TrafficRoutes)
+	services := g.buildServices(endpointMap)
+
+	g.addFilterChains(apiVersion, destinations, endpointMap, info)
+
+	cds, err := g.generateCDS(apiVersion, services, endpointMap)
+	if err != nil {
+		return nil, err
+	}
+	resources.Add(cds...)
+
+	return resources, nil
+}
+
+func (*ExternalServicesGenerator) generateCDS(
+	apiVersion envoy_common.APIVersion,
+	services []string,
+	endpointMap core_xds.EndpointMap,
+) ([]*core_xds.Resource, error) {
+	var resources []*core_xds.Resource
+
+	for _, serviceName := range services {
+		endpoints := endpointMap[serviceName]
+
+		if len(endpoints) == 0 {
+			log.Info("no endpoints for service", "serviceName", serviceName)
+
+			continue
+		}
+
+		clusterBuilder := envoy_clusters.NewClusterBuilder(apiVersion).
+			Configure(envoy_clusters.ProvidedEndpointCluster(
+				serviceName,
+				false, // TODO (bartsmykla)
```

Comment:
and is the plan to resolve this todo?

Reply:
This should more or less be true if dpp is V6.

Reply:
If guess that's available in the xds_context